pkg/api/handlers/slack.go

```diff
@@ -0,0 +1,154 @@
+package handlers
+
+import (
+	"fmt"
+
+	"github.com/gptscript-ai/go-gptscript"
+	"github.com/obot-platform/obot/apiclient/types"
+	"github.com/obot-platform/obot/pkg/api"
+	gatewayTypes "github.com/obot-platform/obot/pkg/gateway/types"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	slackOAuthURL = "https://slack.com/oauth/v2/authorize"
+)
+
+func (p *ProjectsHandler) Configure(req api.Context) error {
+	thread, err := getThreadForScope(req)
+	if err != nil {
+		return err
+	}
+
+	var (
+		input struct {
```

Comment:
Can this be added as a struct in the types. I would call this `SlackReceiver`.  And this same object should be used for create/update.  User writable fields go into the manifest field (just follow the pattern of other types). and status and server controlled fields are outside of the manifest field. This makes it simple to know what can be modified by a user.  In this situation it might be all fields, but still just follow the pattern.